iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/connector/protocol/pipeconsensus/PipeConsensusSyncConnector.java

```diff
@@ -0,0 +1,529 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/** This connector is used for PipeConsensus to transfer queued event. */
+public class PipeConsensusSyncConnector extends IoTDBConnector {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(PipeConsensusSyncConnector.class);
+
+  private static final String PIPE_CONSENSUS_SYNC_CONNECTION_FAILED_FORMAT =
+      "PipeConsensus: syncClient connection to %s:%s failed when %s, because: %s";
+
+  private static final String TABLET_INSERTION_NODE_SCENARIO = "transfer insertionNode tablet";
+
+  private static final String TABLET_RAW_SCENARIO = "transfer raw tablet";
+
+  private static final String TSFILE_SCENARIO = "transfer tsfile";
+
+  private static final String TABLET_BATCH_SCENARIO = "transfer tablet batch";
+
+  private final IClientManager<TEndPoint, SyncPipeConsensusServiceClient>
+      syncRetryAndHandshakeClientManager;
+
+  private final List<TEndPoint> peers;
+
+  private final int thisDataNodeId;
+
+  private String consensusGroupId;
+
+  private PipeConsensusSyncBatchReqBuilder tabletBatchBuilder;
+
+  public PipeConsensusSyncConnector(
+      List<TEndPoint> peers, String consensusGroupId, int thisDataNodeId) {
+    // In PipeConsensus, one pipeConsensusTask corresponds to a pipeConsensusConnector. Thus,
+    // `peers` here actually is a singletonList that contains one peer's TEndPoint. But here we
+    // retain the implementation of list to cope with possible future expansion
+    this.peers = peers;
+    this.consensusGroupId = consensusGroupId;
+    this.thisDataNodeId = thisDataNodeId;
+    this.syncRetryAndHandshakeClientManager =
+        ((PipeConsensus) DataRegionConsensusImpl.getInstance()).getSyncClientManager();
+  }
+
+  @Override
+  public void customize(PipeParameters parameters, PipeConnectorRuntimeConfiguration configuration)
+      throws Exception {
+    super.customize(parameters, configuration);
+    if (isTabletBatchModeEnabled) {
+      tabletBatchBuilder =
+          new PipeConsensusSyncBatchReqBuilder(
+              parameters,
+              new TConsensusGroupId(
+                  TConsensusGroupType.DataRegion, Integer.parseInt(consensusGroupId)),
+              thisDataNodeId);
+    }
+    // currently, tablet batch is false by default in PipeConsensus;
+    isTabletBatchModeEnabled = false;
+  }
+
+  @Override
+  public void handshake() throws Exception {
+    // do nothing
+    // PipeConsensus doesn't need to do handshake, since nodes in same consensusGroup/cluster
+    // usually have same configuration.
+  }
+
+  @Override
+  public void heartbeat() throws Exception {
+    // do nothing
+  }
+
+  @Override
+  public void transfer(TabletInsertionEvent tabletInsertionEvent) throws Exception {
+    // Note: here we don't need to do type judgment here, because PipeConsensus uses DO_NOTHING
```

Comment:
PIPE_CONSENSUS_PROCESSOR

Reply:
If we do not judge here we also should not judge it in asyncConnector.

Reply:
fixed